### ISSUE TITLE
Unable to find main primevideo.com navigation AGAIN

### ISSUE THREAD

**Issue by rob0d** [Sandmann79/xbmc#282]
Hi there,

This seems to be very similar to the issues already reported here. The solution was to update, but I'm running an up-to-date version and still getting it.
I have Libreelec 9.0.1 with Amazon VOD 0.7.0-ter which I upgraded from 0.6.7 (which had the same issue).
I normally have VPN running, but I disabled it to make sure it doesn't interfere.

Full log in debug mode attached.

Any idea what I can try/test/debug to get to the bottom of it?

Thanks

[kodi-amazon.log](https://github.com/Sandmann79/xbmc/files/2949443/kodi-amazon.log)

**Comment by Varstahl** [comment]
It doesn't really tell me much. It's usually either a connection issue or a change in their backend. You could try opening in a browser www.primevideo.com (while logged in), saving the HTML and attaching it to a reply here. 90% of the times though, it's either a connection or VPN related.

**Comment by rob0d** [comment]
Hi Bruno. Thanks for the response.
Had a look at primevideo.com in my browser and strangely enough after logging on I'm getting this:

> PrimeVideo.com is currently unavailable for your account
> As an Amazon Prime member, watch thousands of videos on www.amazon.co.uk. If you live outside of United Kingdom, you may be able to access PrimeVideo.com by updating your account. Learn more about geographic restrictions.

I'm not sure if it's my account as I've enabled subscriptions in the account settings for some prime video channels and they still aren't showing up. I can watch everything directly on prime video via amazon.co.uk, but not via primevideo.com.
Any suggestions? Is this definitely my account problem? Does it work for anyone from UK with UK prime account?

Thanks

**Comment by rob0d** [comment]
OK. Had a chat with Amazon helpdesk people. 
They say:

Deepika: You can access Prime Videos from UK.
Residents within the U.K. can get access to Prime Video from https://www.amazon.co.uk/piv .
Me: ok. so I need to use www.amazon.co.uk/piv rather than primevideo.com?
Deepika: Yes.
Me: only if I were outside of UK I'd be able to use primevideo.com?
Deepika: The Prime Video website ( www.primevideo.com ) is available internationally to residents located outside of the U.K

So basically if I live in the UK I have to use this https://www.amazon.co.uk/piv rather than https://www.primevideo.com/

Would that work with the add-on? Not sure if the pages you scrape look the same or completely different.

**Comment by Varstahl** [comment]
Yes, the addon should already be redirected to the UK version of the site after the login, unless you manually specified to use PrimeVideo in the settings.

In the `Connection > Amazon Account > Region` you can select Auto for automatic detection, which might be a problem if you're using VPN, or manually select United Kingdom. After that, it should work, provided it's not going through the VPN, in which case you won't be able to use primevideo because you're based in the UK, and you can't access the UK version because you're routed through a VPN.

**Comment by rob0d** [comment]
I had it set to UK.
I've completely uninstalled it, completely disabled VPN addon, re-installed Amazon VOD.
Set the region to "Auto" and "United Kingdom", in both cases the same result.

Looking at the code in .../plugin.video.amazon-test/resources/lib/primevideo.py (line 196), I can see you are trying to parse the HTML to find the nav menu". I'm ok with Python, not sure about Kodi Add-ons. Is there a safe way of dumping the HTML it has received to a file so I can have a look at what it sees? Can I just  do open and write to a file in the temp folder?

Thanks

**Comment by Varstahl** [comment]
> Looking at the code in primevideo.py I can see you are trying to parse the HTML to find the nav menu

Yeah, but that's the problem. If you're in UK and you set UK, you **should not** be using primevideo.py, the addon should load amazontld.py. Try to set UK, login again through the addon settings, after you're logged in restart Kodi and see if anything changes.

If you want to poke around, you can check `EntryPoint` in resources/lib/startup.py.

**Comment by rob0d** [comment]
OK. I think I figured it out (after loads of log() statements :).
I had the region set to Auto when I logged on for the first time and it defaulted to PV regardless what I set the region to as it seems that once the user is logged on it's taken from user settings rather then from the config?
Either way. I deleted the cookies. Set region back to UK and logged on again and it's working now.
I'm guessing there may be a little problem with the logic handling the region for logged on users?

**Comment by Varstahl** [comment]
Thought it might be that. When you login the cookies are stored (like the session ones) to avoid saving credentials, potentially dangerous. Along with those data a few more get stored, like a randomly generated device ID and the region your accounts belongs to *as dictated by Amazon's servers*. Surely the first time around, either for some routing or VPN problems, Amazon directed you to primevideo.com. At that point for the addon you're logged in and binded to that, as per Amazon's instructions. That's why I told you to manually select the region and try to log in again (there's the option in the connection settings of the addon).

Glad it's solved, enjoy the shows :)

**Comment by rob0d** [comment]
Thanks for your help and for this add-on!
